### PR TITLE
Recent bookmark count telemetry 

### PR DESCRIPTION
**Pull request by MozillaNoah**
#22075 

### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/metrics.yaml [hunk 1]
```diff
@@ -5893,6 +5893,21 @@ recent_bookmarks:
...
+      Recently Saved section on the home page.
+    bugs:
+      - https://github.com/mozilla-mobile/fenix/issues/22075
+    data_reviews:
+      - https://github.com/mozilla-mobile/fenix/pull/
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - android-probes@mozilla.com
+    expires: "2022-11-01"
```
Comment (MozillaNoah) [inline app/metrics.yaml:5909]: I was just unsure what to make the expiration date now that the telemetry work for MR2 has rolled over into November.

**Comment by MozillaNoah** [comment]
# Request for data collection review form

All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.

1. What questions will you answer with this data?
  * This data will help us understand how users interact with recent bookmarks.
2. Why does Mozilla need to answer these questions? Are there benefits for users? Do we need this information to address product or business requirements? 
  * This will help us understand if the bookmarks section is useful to the user. 
3. What alternative methods did you consider to answer these questions? Why were they not sufficient?
  * There are no other alternatives. 
4. Can current instrumentation answer these questions?
  * No. 
5. List all proposed measurements and indicate the category of data collection for each measurement, using the Firefox data collection categories found on the Mozilla wiki. 
Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.

Measurement Description | Data Collection Category | Tracking Bug #
-- | -- | --
How many recent bookmarks are shown?	| Category 2 - interaction data | https://github.com/mozilla-mobile/fenix/issues/22075

6. Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
  * https://dictionary.telemetry.mozilla.org/apps/fenix 
7. How long will this data be collected? Choose one of the following:
  * One year (ending on November 1st, 2022), with the option to renew at that point. 
8. What populations will you measure?
  * All channels, all locales, all countries 
9. If this data collection is default on, what is the opt-out mechanism for users?
  * Default Glean SDK opt-out mechanism. 
10. Please provide a general description of how you will analyze this data.
  * Glean and Amplitude. 
11. Where do you intend to share the results of your analysis?
  *  Only on Glean, Amplitude, and with mobile teams. 
12. Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?
   * No third-party tools.

File: app/metrics.yaml (hunk 1 above)
Comment (rocketsroger) [reply app/metrics.yaml:5909]: Looks good to me

File: app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt [hunk 2]
```diff
@@ -823,6 +823,10 @@ private val Event.wrapper: EventWrapper<*>?
             { RecentBookmarks.shown.record(it) }
         )
 
+        is Event.RecentBookmarkCount -> EventWrapper<NoExtraKeys>(
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt:826]: This won't work.  Take a look at `HistoryRecentSearchesTapped` for when you have value with your event.

**Review by rocketsroger — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt (hunk 2 above)
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/components/metrics/GleanMetricsService.kt:826]: Nice, quantity seems to be a good fit here.

**Review by rocketsroger — APPROVED** [review]
🚢

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v95.0.0

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v94.0.0

**Pull request merged**